src/platform/linux/inputtino.cpp

```diff
@@ -0,0 +1,648 @@
+          if (adjusted_angle > 90) {
+            adjusted_angle -= 360;
+          }
+          else if (adjusted_angle < -90) {
+            adjusted_angle += 360;
+          }
+          (*raw->touch).place_finger(touch.pointerId, touch.x, touch.y, touch.pressureOrDistance, adjusted_angle);
+          break;
+        }
+        case LI_TOUCH_EVENT_CANCEL:
+        case LI_TOUCH_EVENT_UP:
+        case LI_TOUCH_EVENT_HOVER_LEAVE: {
+          (*raw->touch).release_finger(touch.pointerId);
+          break;
+        }
+          // TODO: LI_TOUCH_EVENT_CANCEL_ALL
+      }
+    }
+  }
+
+  static inline float
+  deg2rad(float degree) {
+    return degree * (M_PI / 180.f);
+  }
+
+  void
+  pen(client_input_t *input, const touch_port_t &touch_port, const pen_input_t &pen) {
+    auto raw = (client_input_raw_t *) input;
+    if (raw->pen) {
+      // First set the buttons
+      (*raw->pen).set_btn(inputtino::PenTablet::PRIMARY, pen.penButtons & LI_PEN_BUTTON_PRIMARY);
+      (*raw->pen).set_btn(inputtino::PenTablet::SECONDARY, pen.penButtons & LI_PEN_BUTTON_SECONDARY);
+      (*raw->pen).set_btn(inputtino::PenTablet::TERTIARY, pen.penButtons & LI_PEN_BUTTON_TERTIARY);
+
+      // Set the tool
+      inputtino::PenTablet::TOOL_TYPE tool;
+      switch (pen.toolType) {
+        case LI_TOOL_TYPE_PEN:
+          tool = inputtino::PenTablet::PEN;
+          break;
+        case LI_TOOL_TYPE_ERASER:
+          tool = inputtino::PenTablet::ERASER;
+          break;
+        default:
+          tool = inputtino::PenTablet::SAME_AS_BEFORE;
+          break;
+      }
+
+      // Normalize rotation value to 0-359 degree range
+      auto rotation = pen.rotation;
+      if (rotation != LI_ROT_UNKNOWN) {
+        rotation %= 360;
+      }
+
+      // Here we receive:
+      //  - Rotation: degrees from vertical in Y dimension (parallel to screen, 0..360)
+      //  - Tilt: degrees from vertical in Z dimension (perpendicular to screen, 0..90)
+      float tilt_x = 0;
+      float tilt_y = 0;
+      // Convert polar coordinates into Y tilt angles
+      if (pen.tilt != LI_TILT_UNKNOWN && rotation != LI_ROT_UNKNOWN) {
+        auto rotation_rads = deg2rad(rotation);
+        auto tilt_rads = deg2rad(pen.tilt);
+        auto r = std::sin(tilt_rads);
+        auto z = std::cos(tilt_rads);
+
+        tilt_x = std::atan2(std::sin(-rotation_rads) * r, z) * 180.f / M_PI;
+        tilt_y = std::atan2(std::cos(-rotation_rads) * r, z) * 180.f / M_PI;
+      }
+
+      (*raw->pen).place_tool(tool,
+        pen.x,
+        pen.y,
+        pen.eventType == LI_TOUCH_EVENT_DOWN ? pen.pressureOrDistance : -1,
+        pen.eventType == LI_TOUCH_EVENT_HOVER ? pen.pressureOrDistance : -1,
+        tilt_x,
+        tilt_y);
+    }
+  }
+
+  enum ControllerType {
+    XboxOneWired,
+    DualSenseWired,
+    SwitchProWired
+  };
+
+  int
+  alloc_gamepad(input_t &input, const gamepad_id_t &id, const gamepad_arrival_t &metadata, feedback_queue_t feedback_queue) {
```

Comment:
The gamepad options will need be exposed in the config ui.

https://github.com/LizardByte/Sunshine/blob/8316f44e10f3f6ef27fc27b3ef6e69f359b1b667/src_assets/common/assets/web/config.html#L141C1-L150C15

Can probably become (untested).

```html
        <!-- Emulated Gamepad Type -->
        <div class="mb-3" v-if="config.controller === 'enabled' && (platform === 'windows' || platform === 'linux')">
          <label for="gamepad" class="form-label">Emulated Gamepad Type</label>
          <select id="gamepad" class="form-select" v-model="config.gamepad">
            <option value="auto">Automatic</option>
            <option value="ds4">DS4 (PS4)</option>
            <option value="switch" v-if="platform === 'linux'">Nintendo Pro (Switch)</option>
            <option value="x360">X360 (Xbox 360)</option>
          </select>
          <div class="form-text">Choose which type of gamepad to emulate on the host</div>
        </div>
```